contracts/governance/extensions/GovernorOverrideDelegateVote.sol

```diff
@@ -0,0 +1,194 @@
+
+    constructor(VotesOverridable tokenAddress) GovernorVotes(tokenAddress) {}
+
+    /**
+     * @dev Fetch the past delegate for an `account` at a given `timepoint` from the token.
+     */
+    function _getPastDelegate(address account, uint256 timepoint) internal view virtual returns (address) {
+        return VotesOverridable(address(token())).getPastDelegate(account, timepoint);
+    }
+
+    /**
+     * @dev Fetch the past `balanceOf` for an `account` at a given `timepoint` from the token.
+     */
+    function _getPastBalanceOf(address account, uint256 timepoint) internal view virtual returns (uint256) {
+        return VotesOverridable(address(token())).getPastBalanceOf(account, timepoint);
+    }
+
+    /**
+     * @dev See {IGovernor-COUNTING_MODE}.
+     */
+    // solhint-disable-next-line func-name-mixedcase
+    function COUNTING_MODE() public pure virtual override returns (string memory) {
+        return "support=bravo,override&quorum=for,abstain&params=override";
+    }
+
+    /**
+     * @dev See {IGovernor-hasVoted}.
+     */
+    function hasVoted(uint256 proposalId, address account) public view virtual override returns (bool) {
+        return _proposalVotes[proposalId].voteReceipt[account].hasVoted;
+    }
+
+    /**
+     * @dev Check if an `account` has overridden their delegate for a proposal.
+     */
+    function hasVotedOverride(uint256 proposalId, address account) public view virtual returns (bool) {
+        return _proposalVotes[proposalId].overrideVoteReceipt[account].hasVoted;
+    }
+
+    /**
+     * @dev Accessor to the internal vote counts.
+     */
+    function proposalVotes(
+        uint256 proposalId
+    ) public view virtual returns (uint256 againstVotes, uint256 forVotes, uint256 abstainVotes) {
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+        return (proposalVote.againstVotes, proposalVote.forVotes, proposalVote.abstainVotes);
+    }
+
+    /**
+     * @dev See {Governor-_quorumReached}.
+     */
+    function _quorumReached(uint256 proposalId) internal view virtual override returns (bool) {
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+
+        return quorum(proposalSnapshot(proposalId)) <= proposalVote.forVotes + proposalVote.abstainVotes;
+    }
+
+    /**
+     * @dev See {Governor-_voteSucceeded}. In this module, the forVotes must be strictly over the againstVotes.
+     */
+    function _voteSucceeded(uint256 proposalId) internal view virtual override returns (bool) {
+        ProposalVote storage proposalVote = _proposalVotes[proposalId];
+
+        return proposalVote.forVotes > proposalVote.againstVotes;
+    }
+
+    function _countVote(
+        uint256 proposalId,
+        address account,
+        uint8 support,
+        uint256 totalWeight,
+        bytes memory params
+    ) internal virtual override returns (uint256) {
+        if (keccak256(params) == keccak256(hex"23b70c8d0000000000000000000000000000000000000000")) {
```

Comment:
What does "0x23b70c8d" represent. Can we do something more readable ?

Reply:
I accidently removed a comment explaining this--sorry about that.

```
// If the params are the selector `0x23b70c8d` (keccak("override")[0:8]) followed by address 0, it is an override vote.
```

The formatting of the params and how the override vote is communicated to indexers is something that definitely could be further discussed.

Reply:
Good idea to use something like a selector to identify the type of parameters. I agree it requires more discussion but in GovernorCountingFractional [we used `support == 255` as a discriminator](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/c1d6ad5a307945c7825cb3c4c911f2fd017c8725/contracts/governance/extensions/GovernorCountingFractional.sol#L34).

If we were to use a selector, then I wouldn't compute it in runtime, and instead save a constant for it:

```solidity
/// keccak256(abi.encodePacked(bytes4(keccak256('override')), bytes20(address(0x0000000000000000000000000000000000000000))))
bytes32 constant DISCRIMINATOR = 0x4b59cd2c312bedbbf68f0e4aa2fcd2c8476080b4bce6a8cd9a550e8834b2219c
```

Reply:
Following discussion  with Tally, override should ahve its own public facing function(s). This removes the need for params checks here.